### ISSUE TITLE
Layout/RedundantLineBreak modifies multiline strings

### ISSUE THREAD

**Issue by naveg** [rubocop/rubocop#11949]
`Layout/RedundantLineBreak` is not respecting multiline strings. Consider this example, `foo.rb`:

```ruby
# frozen_string_literal: true

foo = 'this is
  a string
 with    some
     funky
  but valid
   whitespace
'

puts foo
```

and this rubocop config `.rubocop.yml`:

```
AllCops:
  NewCops: enable

Layout/RedundantLineBreak:
  Enabled: true
```

## Expected behavior

This code is valid Ruby, and Ruby respects the whitespace as written. Running the program results in this output:

```
❯ ruby foo.rb
this is
  a string
 with    some
     funky
  but valid
   whitespace
```

As such, RuboCop should not flag any offenses.

## Actual behavior

RuboCop flags a `Layout/RedundantLineBreak` offense, and modifies the code. The resulting program behaves differently.

```
❯ rubocop -a foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:3:1: C: [Corrected] Layout/RedundantLineBreak: Redundant line break detected.
foo = 'this is ...
^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected

❯ cat foo.rb
# frozen_string_literal: true

foo = 'this is a string with    some funky but valid whitespace '

puts foo

❯ ruby foo.rb
this is a string with    some funky but valid whitespace
```

## RuboCop version

```
❯ rubocop -V
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.28.1, running on ruby 3.2.0) [arm64-darwin22]
❯ ruby --version
ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin22]
```
